crypto/conf/conf_def.c

```diff
@@ -9,29 +9,40 @@
 
 /* Part of the code in here was originally in conf.c, which is now removed */
 
+#include "e_os.h"
 #include <stdio.h>
 #include <string.h>
+#include <sys/stat.h>
```

Comment:
Hmmm.... stat is kind of POSIX thing and there is such thing as OPENSSL_NO_POSIX_IO... I mean I suppose it would be appropriate to use it even here... Which in turn asks for question how to handle the cases that use stat. Would it be appropriate to simply attempt to open it as if it was a directory and see how it works out? Or maybe whole feature should be unavailable without POSIX?

Reply:
Maybe I could make the directory loading unsupported on OPENSSL_NO_POSIX_IO systems?

Reply:
BTW, store/loader_file.c also calls stat() without any consideration for non-POSIX.

Reply:
Yes, if no POSIX then you can't include a directory.  That's a good trade-off.

Reply:
Or I can just try first to read the include path as directory and if it fails try just BIO_new_file(). That's also possibility. Which one do you prefer?

Reply:
I would not do that; on pre-BSD4.2 you can open(2) directories :)  More usefully, don't try something that you expect to usually fail.

Reply:
> I would not do that; on pre-BSD4.2 you can open(2) directories

But that's not what is suggested. What is suggested is equivalent of opendir first and open then. It does require that you can rely on "not a directory" error code, and it might be and seems to be platform specific thing. In other words, I'd too say "no include without POSIX".

Reply:
No include of directories, or no include support at all without POSIX?

Reply:
If it's not POSIX, don't try to include directories.  That seems simplest way out of this.